src/gui/transferlistmodel.cpp

```diff
@@ -158,37 +182,175 @@ QVariant TransferListModel::headerData(int section, Qt::Orientation orientation,
     return {};
 }
 
-QVariant TransferListModel::data(const QModelIndex &index, const int role) const
+QString TransferListModel::displayValue(const BitTorrent::TorrentHandle *torrent, int column) const
 {
-    if (!index.isValid()) return {};
-
-    const BitTorrent::TorrentHandle *torrent = m_torrentList.value(index.row());
-    if (!torrent) return {};
-
-    if ((role == Qt::DecorationRole) && (index.column() == TR_NAME))
-        return getIconByState(torrent->state());
-
-    if (role == Qt::ForegroundRole)
-        return getColorByState(torrent->state());
+    const bool isHideState = (Preferences::instance()->getHideZeroComboValues() == 1)
+            && (torrent->state() == BitTorrent::TorrentState::PausedDownloading); // paused torrents only
+    const bool hideValues = Preferences::instance()->getHideZeroValues() && isHideState;
+
+    const auto availabilityString = [hideValues](const qreal value)
+    {
+        return ((value <= 0) && hideValues)
+                ? QString {} : Utils::String::fromDouble(value, 3);
+    };
+
+    const auto unitString = [hideValues](const qint64 value, const bool isSpeedUnit = false)
+    {
+        return ((value == 0) && hideValues)
+                ? QString {} : Utils::Misc::friendlyUnit(value, isSpeedUnit);
+    };
+
+    const auto limitString = [hideValues](const qint64 value)
+    {
+        return ((value == 0) && hideValues)
+                ? QString {}
+                : (value > 0)
+                  ? Utils::Misc::friendlyUnit(value, true)
+                  : QString::fromUtf8(C_INFINITY);
+    };
+
+    const auto amountString = [hideValues](const qint64 value, const qint64 total)
+    {
+        return ((value == 0) && (total == 0) && hideValues)
+                ? QString {} : QString::number(value) + " (" + QString::number(total) + ')';
+    };
+
+    const auto ratioString = [hideValues](const qreal value)
+    {
+        return ((value <= 0) && hideValues)
+                ? QString {}
+                : ((static_cast<int>(value) == -1) || (value > BitTorrent::TorrentHandle::MAX_RATIO))
+                  ? QString::fromUtf8(C_INFINITY) : Utils::String::fromDouble(value, 2);
+    };
+
+    const auto queuePositionString = [](const qint64 value)
+    {
+        return (value > 0) ? QString::number(value) : "*";
+    };
+
+    const auto lastActivityString = [hideValues](qint64 value)
+    {
+        if (hideValues && ((value < 0) || (value >= MAX_ETA)))
+            return QString {};
```

Comment:
>|| (value >= MAX_ETA)))

Does this mean it will show nothing if torrent was inactive for >100 days?

Reply:
I didn't touch this logic. I just moved things into appropriated place.

Reply:
oops